server/controllers/finance/journal/index.js

```diff
@@ -36,6 +40,64 @@ exports.journalEntryList = journalEntryList;
 
 exports.editTransaction = editTransaction;
 exports.count = count;
+exports.refenreceTransaction=refenreceTransaction;
+
+
+
+
+
+//loading identifiers keyies, used for defining the table name
+const identifiersIndex = {};
+function indexIdentifiers() {
+  _.forEach(identifiers, (entity) => {
+    identifiersIndex[entity.key] = entity;
+  });
+}
+
+// API GET /journal/refenrece_transaction/:codeRef/:language'
+
+//This function render a report in the browser
+//It takes a transaction reference code as paramter and the language
+//The reference code is a combination of table_key.project_abbr.reference
+//The table name is variable, it can be :invoice, cash or voucher
+
+function refenreceTransaction(req,res,next){
+   
+  //loading identifiers keyies, used for defining the table name
+  indexIdentifiers();
+
+  //reports urls
+  var reportsTypes={
+    VO:'/reports/finance/vouchers/',
+    IV:'/reports/finance/invoices/',
+    CP:'/reports/finance/cash/'
+  };
+
+  var codeRef =  req.params.codeRef.split(".");
+  const language= req.params.language;
+
+  const code = codeRef[0];
+  const project_name=codeRef[1];
+  const refence=codeRef[2];
+  const documentDefinition = identifiersIndex[code];
+   
+   const query = `
+    SELECT BUID(uuid) as uuid FROM ${documentDefinition.table}  tb, project p 
+    WHERE tb.project_id=p.id AND p.abbr="${project_name}" AND tb.reference="${refence}"
+  `;
+  
+  // search for full UUID
+  db.one(query)
```

Comment:
It would be nicer to follow the repository standards for indentation.  We generally write our code like this:
```js
db.one(query)
  .then(row => {
     console.log(row.id);
     return 1;
  })
  .then(integer => {
     // do something else in this block
   })
   .catch(err => {
      // include some error handling code
   })
   .done();
```